packages/firestore/test/unit/local/remote_document_cache.test.ts

```diff
@@ -312,14 +313,57 @@ function genericRemoteDocumentCacheTests(
     );
 
     const query = new Query(path('b'));
-    const matchingDocs = await cache.getDocumentsMatchingQuery(query);
+    const matchingDocs = await cache.getDocumentsMatchingQuery(
+      query,
+      SnapshotVersion.MIN
+    );
 
     assertMatches(
       [doc('b/1', VERSION, DOC_DATA), doc('b/2', VERSION, DOC_DATA)],
       matchingDocs
     );
   });
 
+  it('can get documents matching query by read time', async () => {
+    await cache.addEntries(
+      [doc('b/old', 1, DOC_DATA)],
+      /* readTime= */ version(11)
+    );
+    await cache.addEntries(
+      [doc('b/current', 2, DOC_DATA)],
+      /* readTime= */ version(12)
+    );
+    await cache.addEntries(
+      [doc('b/new', 3, DOC_DATA)],
+      /* readTime= */ version(13)
+    );
+
+    const query = new Query(path('b'));
+    const matchingDocs = await cache.getDocumentsMatchingQuery(
+      query,
+      /* sinceReadTime= */ version(12)
+    );
+    assertMatches([doc('b/new', 3, DOC_DATA)], matchingDocs);
+  });
+
+  it('query matching uses read time rather than update time', async () => {
+    await cache.addEntries(
+      [doc('b/old', 1, DOC_DATA)],
+      /* readTime= */ version(2)
+    );
+    await cache.addEntries(
+      [doc('b/new', 2, DOC_DATA)],
+      /* readTime= */ version(1)
```

Comment:
Reading versions that haven't been written yet!  Impressive.

(I assume this is intentional, though I'm also not sure it's a valuable test.  I can't immediately think of a bug that this test would catch that the previous one wouldn't).

Reply:
Yeah, this might have been more valuable on Android since I switched the implementation from updateTime to readTime. I'll keep it for consistency.